controllers/sessionController.js

```diff
@@ -0,0 +1,51 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+const csrf = require("host-csrf");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res, next) => {
+  if (req.body.password != req.body.password1) {
+    req.flash("error", "The passwords entered do not match.");
+    res.render("register");
```

Comment:
You must pass the error as the second parameter od the render method here. It's needed for the error to be displayed. Also, you must add a return here unless the application continues to the next line.
Example:
```
 return res.render("register", {  errors: req.flash("error")});
```